cmd/ansible-operator/main.go

```diff
@@ -37,18 +42,84 @@ type config struct {
 
 func main() {
 	printVersion()
+	done := make(chan error)
+
+	// start the proxy
+	go runProxy("localhost", 8888, done)
+
+	// start the operator
+	go runSDK(done)
+
+	// wait for either to finish
+	err := <-done
+	if err == nil {
+		logrus.Info("Exiting")
+	} else {
+		logrus.Fatal(err.Error())
+	}
+}
+
+// readConfig reads the operator's config file at /opt/ansible/config.yaml
+func readConfig() ([]config, error) {
+	b, err := ioutil.ReadFile("/opt/ansible/config.yaml")
+	if err != nil {
+		logrus.Fatalf("failed to get config file %v", err)
+	}
+	c := []config{}
+	err = yaml.Unmarshal(b, &c)
+	if err != nil {
+		logrus.Fatalf("failed to unmarshal config %v", err)
+	}
+	return c, nil
+}
+
+func registerGVK(gvk schema.GroupVersionKind) {
+	schemeBuilder := k8sruntime.NewSchemeBuilder(func(s *k8sruntime.Scheme) error {
+		s.AddKnownTypeWithName(gvk, &unstructured.Unstructured{})
+		return nil
+	})
+	k8sutil.AddToSDKScheme(schemeBuilder.AddToScheme)
+}
+
+// getConfig returns a rest.Config object. It tries to find an in-cluster
+// configuration, and if not found, falls back to looking for a kubeconfig at
+// file path ~/.kube/config
+func getConfig() (*rest.Config, error) {
```

Comment:
I think that there is already an operator sdk exposed way to get this config.

Reply:
Should work fine, though it does delegate in the opposite order, and doesn't seem to have a default location for the kubeconfig.